packages/@react-aria/menu/stories/useMenu.stories.tsx

```diff
@@ -105,7 +105,7 @@ function MenuPopup(props) {
   return (
     <FocusScope restoreFocus>
       <div {...overlayProps} ref={overlayRef}>
-        <DismissButton onDismiss={props.onClose} />
+        <DismissButton onDismiss={props.onClose} style={{transform: 'translate(0, -100%)'}} />
```

Comment:
It's not really a problem with `VisuallyHidden`. It's a problem with positioning the DismissButton so that it does not overlap behind a MenuItem in the overlay, and it depends on where the DismissButton is rendered in the DOM, before or after the Menu.

Reply:
Related comment thread: https://github.com/adobe/react-spectrum/pull/3048#discussion_r869473386